components/launcher/src/server/handlers/restart.rs

```diff
@@ -25,7 +24,7 @@ impl Handler for RestartHandler {
     type Reply = protocol::SpawnOk;
 
     fn handle(msg: Self::Message, services: &mut ServiceTable) -> HandleResult<Self::Reply> {
-        let mut service = match services.remove(msg.get_pid() as Pid) {
+        let mut service = match services.remove(msg.get_pid() as u32) {
```

Comment:
I like descriptive types as much as the next person, but [`std::process::Child::id()`](http://doc.rust-lang.org/1.29.1/std/process/struct.Child.html#method.id) returns a `u32`, so I think it's preferable to align with the stdlib. This ripples out in a number of places, but also requires some casts since certain `libc` calls take `i32`.